File: API/Helpers/UserParams.cs
namespace API.Helpers;
public class UserParams : PaginationParams
{
    public string? CurrentUsername { get; set; }
    public string? Gender { get; set; }

    public int MinAge { get; init; } = 15;
    public int MaxAge { get; init; } = 100;

    public string OrderBy { get; init; } = "lastActive";
}

File: rumble-client/src/app/models/userParams.ts
import { User } from './user';

export class UserParams {
  gender: string = '';
  minAge = 15;
  maxAge = 100;
  pageNumber = 1;
  pageSize = 10;
  orderBy = 'lastActive';

  constructor(user: User) {
    this.gender = user.gender === 'Male' ? 'Female' : 'Male';
  }
}
